hex/lib/dependabot/hex/credential_helpers.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Hex
+    module CredentialHelpers
+      def self.hex_credentials(credentials)
+        organization_credentials(credentials) + repo_credentials(credentials)
+      end
+
+      def self.organization_credentials(credentials)
+        credentials.
+          select { |cred| cred["type"] == "hex_organization" }.
+          flat_map { |cred| cred.slice("type", "organization", "token").values }
+      end
+
+      def self.repo_credentials(credentials)
+        # Credentials are serialized as an array that may not have optional fields. Using a
+        # default ensures that the array is always the same length, even if values are empty.
+        defaults = { "url" => "", "token" => "", "public_key_fingerprint" => "" }
+        keys = %w(type repo url token public_key_fingerprint)
```

Comment:
Internally, we're setting up the registry entry to use `auth-key`, which is consistent with the Hex docs.

```suggestion
        defaults = { "url" => "", "auth_key" => "", "public_key_fingerprint" => "" }
        keys = %w(type repo url auth_key public_key_fingerprint)
```

---

I know the Hex docs use `fetch-public-key`, but a teammate and I found that somewhat harder to reason about, so we're adopting your term here (`public_key_fingerprint`) for that entry 😄